src/watertap_contrib/seto/analysis/multiperiod/vagmd_batch/VAGMD_batch_flowsheet.py

```diff
@@ -0,0 +1,427 @@
+# Pyomo imports
+from pyomo.environ import (
+    ConcreteModel,
+    Var,
+    Objective,
+    Param,
+    Expression,
+    Constraint,
+    Block,
+    log10,
+    TransformationFactory,
+    assert_optimal_termination,
+    value,
+    units as pyunits,
+)
+from pyomo.util.calc_var_value import calculate_variable_from_constraint
+
+# IDAES imports
+from idaes.core.util.model_statistics import degrees_of_freedom
+import idaes.core.util.scaling as iscale
+from idaes.core import FlowsheetBlock
+from idaes.core.solvers.get_solver import get_solver
+import idaes.logger as idaeslog
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    unscaled_variables_generator,
+    badly_scaled_var_generator,
+)
+
+# WaterTAP imports
+from watertap.property_models.seawater_prop_pack import SeawaterParameterBlock
+from watertap.property_models.water_prop_pack import WaterParameterBlock
+from watertap_contrib.seto.unit_models.surrogate import VAGMDSurrogate
+
+
+def build_vagmd_flowsheet(
+    m=None,
+    dt=None,
+    system_capacity=2000,  # m3/day
+    feed_flow_rate=600,
+    evap_inlet_temp=80,
+    cond_inlet_temp=25,
+    feed_temp=25,
+    feed_salinity=35,
+    recovery_ratio=0.5,
+    initial_batch_volume=50,
+    module_type="AS7C1.5L",
+    cooling_system_type="closed",
+    high_brine_salinity=False,  # True if brine salinity > 175.3 g/L
+    cooling_inlet_temp=25,  # Not required when cooling system type is "closed"
+):
+    """
+    This function builds a unit model for a certain time period
+
+    Returns:
+        object: A Pyomo concrete optimization model and flowsheet
+    """
+    if m is None:
+        m = ConcreteModel()
+
+    m.fs = FlowsheetBlock(dynamic=False)
+    m.fs.seawater_properties = SeawaterParameterBlock()
+    m.fs.water_properties = WaterParameterBlock()
+
+    m.fs.vagmd = VAGMDSurrogate(
+        property_package_seawater=m.fs.seawater_properties,
+        property_package_water=m.fs.water_properties,
+        module_type=module_type,
+        high_brine_salinity=high_brine_salinity,
+        cooling_system_type=cooling_system_type,
+    )
+
+    # Run helper function to determine the salinity mode
+    (
+        feed_flow_rate,
+        evap_inlet_temp,
+        cond_inlet_temp,
+        cooling_system_type,
+    ) = m.fs.vagmd._determine_salinity_mode(
+        feed_flow_rate,
+        evap_inlet_temp,
+        cond_inlet_temp,
+        module_type,
+        high_brine_salinity,
+        cooling_system_type,
+    )
+
+    # Specify system capacity
+    m.fs.vagmd.system_capacity.fix(system_capacity)
+    # Specify feed flow state properties
+    m.fs.vagmd.feed_props.calculate_state(
```

Comment:
You should scale the state variables, then run calculate_scaling_factors, then run calculate_state. Not doing so can lead to issues. See this WaterTAP issue for more background https://github.com/watertap-org/watertap/issues/1123

In any case, this can be reserved for a subsequent PR and tracked in an issue for refinment, so long as checks pass.